components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/applications/ApplicationUtils.java

```diff
@@ -191,52 +128,56 @@ public static StringBuilder getTextPayload (String appId, String groupName, Stri
         return payloadBuilder;
     }
 
-    public static PayloadData createPayload(String appId, String groupName, CartridgeInfo cartridgeInfo, String subscriptionKey, int tenantId, String clusterId,
+    public static PayloadData createPayload(String appId, String groupName, Cartridge cartridge, String subscriptionKey, int tenantId, String clusterId,
```

Comment:
The code would have looked much cleaner if an instance of an appropriate bean class was used to wrap all these different pieces of parameters and passed it into the method as an argument.